packages/E2ETest/run_wdio.js

```diff
@@ -100,20 +114,42 @@ function parseLogs() {
   return names;
 }
 
+function PrintFailedTests(ft) {
+  for (let i = 0; i < Object.keys(ft).length; i++) {
```

Comment:
To iterate through object keys of an object use a `for in` loop. To iterate through elements of an iterable (e.g. array) use a `for of` loop.

Reply:
the best way to remember whether to use `for...in` or `for...of` in JavaScript is to think what common sense would be (element in an array, property of an object) and do the exact opposite